### ISSUE TITLE
Wrong position reported for unicode strings

### ISSUE THREAD

**Issue by nijel** [cweiske/php-sqllint#14]
The error reporting wrongly reports error position in unicode strings.

The problem is that the parser operates on chars (which can be multibyte in utf-8), while the error displaying code on bytes:

https://github.com/cweiske/php-sqllint/blob/07ca0a3244fa872f2f9fc25adf3562a325d757a1/src/phpsqllint/Cli.php#L96

You can safely use mb_* functions as the sql parser uses them as well (it depends on [symfony/polyfill-mbstring](https://packagist.org/packages/symfony/polyfill-mbstring) in case mbstring extension ins not installed).

Original reported as https://github.com/phpmyadmin/sql-parser/issues/177:

sample code:
``` mysql
SELECT 'выбор типа';
SELECT 1
#INSERT INTO tbl2 (name) VALUES (
	SELECT 1;
#);
```
(missed semicolon after second SELECT)

Currently parser return error:
```
test.sql (line 3, col 33): This type of clause was previously parsed. (near: SELECT)
#INSERT INTO tbl2 (name) VALUES (
                                ^
```

Expected: commented line can not has error pointer.

**Hint**: problem in UTF-8 russian string.

**Comment by cweiske** [comment]
php-sqllint does not show any code nor the code position; it only says where the error was:

```
Checking SQL syntax of tests/files/unicode-error.sql
 Line 1, col 13 at "SELECT": This type of clause was previously parsed.
 Line 1, col 0 at "": An expression was expected.
```

I tried it with several unicode characters, and the positions are always correct.

**Comment by cweiske** [comment]
The example file currently reports

```
Checking SQL syntax of tests/files/unicode-error.sql
 Line 4, col 0 at "SELECT": This type of clause was previously parsed.
```

but it should report
```
Checking SQL syntax of tests/files/unicode-error.sql
 Line 4, col 9 at "SELECT": This type of clause was previously parsed.
```
